src/block.rs

```diff
@@ -170,46 +170,185 @@ impl BlockBuilder {
+                0,
+                Some(0),
+                Some(0),
+            ),
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key1"),
+                ValueDeletable::Value(Bytes::copy_from_slice(b"value1")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key2"),
+                ValueDeletable::Value(Bytes::copy_from_slice(b"value2")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+        ],
+        expected_size: 130,
+    })]
+    #[case(BlockTestCase {
+        name: "block_with_tombstone",
+        entries: vec![
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key1"),
+                ValueDeletable::Value(Bytes::copy_from_slice(b"value1")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key2"),
+                ValueDeletable::Tombstone,
+                0,
+                Some(0),
+                Some(0),
+            ),
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key3"),
+                ValueDeletable::Value(Bytes::copy_from_slice(b"value3")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+        ],
+        expected_size: 121,
+    })]
+    #[case(BlockTestCase {
+        name: "block_with_merge",
+        entries: vec![
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key1"),
+                ValueDeletable::Value(Bytes::copy_from_slice(b"value1")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key1"),
+                ValueDeletable::Merge(Bytes::copy_from_slice(b"value1")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+            RowEntry::new(
+                Bytes::copy_from_slice(b"key2"),
+                ValueDeletable::Value(Bytes::copy_from_slice(b"value2")),
+                0,
+                Some(0),
+                Some(0),
+            ),
+        ],
+        expected_size: 130,
+    })]
+    fn test_block(#[case] test_case: BlockTestCase) {
+        let block = build_block(&test_case);
+        let encoded = block.encode();
+        let decoded = Block::decode(encoded.clone());
+        let block_data = &block.data;
+        let block_offsets = &block.offsets;
+        // Decode the block data using offsets and validate each decoded entry
+        assert_decoded_entries(block_data, block_offsets, &test_case.entries)
+            .expect("Decoding error");
+
+        assert_eq!(block_data, &decoded.data);
+        assert_eq!(block_offsets, &decoded.offsets);
+        assert_eq!(block.size(), test_case.expected_size);
```

Comment:
Do we need to check block size if debug snapshot is validating a byte-for-byte match?

Reply:
Yes, we can remove the size check. My reasoning is that we need to test three core aspects: block data, offsets, and size. Including the size check allows us to catch potential failures earlier in the process, before reaching the final step of asserting the debug snapshot.